### ISSUE TITLE
Java submission runs over TL on Cerberus

### ISSUE THREAD

**Issue by cheesycake** [DMOJ/judge#138]
```
Attaching to process ID 12828, please wait...
Debugger attached successfully.
Server compiler detected.
JVM version is 24.79-b02
Deadlock Detection:

No deadlocks found.

Thread 20: (state = IN_JAVA)
 - java.lang.String.<init>(char[], int, int) @bci=43 (Compiled frame; information may be imprecise)
 - java.io.BufferedReader.readLine(boolean) @bci=208 (Compiled frame)
 - java.io.BufferedReader.readLine() @bci=2 (Compiled frame)
 - testss.main(java.lang.String[]) @bci=43, line=13 (Compiled frame)
 - sun.reflect.NativeMethodAccessorImpl.invoke0(java.lang.reflect.Method, java.lang.Object, java.lang.Object[]) @bci=0 (Interpreted frame)
 - sun.reflect.NativeMethodAccessorImpl.invoke(java.lang.Object, java.lang.Object[]) @bci=87 (Interpreted frame)
 - sun.reflect.DelegatingMethodAccessorImpl.invoke(java.lang.Object, java.lang.Object[]) @bci=6 (Interpreted frame)
 - java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) @bci=57 (Interpreted frame)
 - ca.dmoj.java.SubmissionThread.run() @bci=53, line=27 (Interpreted frame)


Thread 19: (state = BLOCKED)
 - java.lang.Thread.sleep(long) @bci=0 (Interpreted frame)
 - ca.dmoj.java.ShockerThread.run() @bci=4, line=22 (Interpreted frame)


Thread 14: (state = BLOCKED)


Thread 13: (state = BLOCKED)


Thread 12: (state = BLOCKED)
 - java.lang.Object.wait(long) @bci=0 (Interpreted frame)
 - java.lang.ref.ReferenceQueue.remove(long) @bci=44 (Interpreted frame)
 - java.lang.ref.ReferenceQueue.remove() @bci=2 (Interpreted frame)
 - java.lang.ref.Finalizer$FinalizerThread.run() @bci=36 (Interpreted frame)


Thread 11: (state = BLOCKED)
 - java.lang.Object.wait(long) @bci=0 (Interpreted frame)
 - java.lang.Object.wait() @bci=2, line=503 (Interpreted frame)
 - java.lang.ref.Reference$ReferenceHandler.run() @bci=46 (Interpreted frame)


Thread 1: (state = BLOCKED)
 - java.lang.Object.wait(long) @bci=0 (Interpreted frame)
 - java.lang.Thread.join(long) @bci=38 (Interpreted frame)
 - java.lang.Thread.join() @bci=2 (Interpreted frame)
 - ca.dmoj.java.JavaSafeExecutor.main(java.lang.String[]) @bci=300, line=167 (Interpreted frame)
```

Java Version:
```
java version "1.7.0_79"
Java(TM) SE Runtime Environment (build 1.7.0_79-b15)
Java HotSpot(TM) 64-Bit Server VM (build 24.79-b02, mixed mode)
```

**Comment by Xyene** [comment]
Additional tidbit of info: the JVM is hung, `jstack` had to be run with `-F` to produce any output.

**Comment by cheesycake** [comment]
Result of 4th case:
```
INFO 2016-01-11 23:17:21,171 packet Submission test case #4: 169951
Test case 4
        5892.781250 seconds (real)
        5892.781250 seconds (debugged)
        61.85 mb (63336.0 kb)
        WA
        TLE
```
Second jstack -F on 5th case:
```
C:\Program Files\Java\jdk1.7.0_79\bin>tasklist | findstr java
java.exe                     10528 Console                   10     59,892 K

C:\Program Files\Java\jdk1.7.0_79\bin>jstack.exe -F 10528
Attaching to process ID 10528, please wait...
Debugger attached successfully.
Server compiler detected.
JVM version is 24.79-b02
Deadlock Detection:

No deadlocks found.

Thread 20: (state = IN_JAVA)
 - testss.main(java.lang.String[]) @bci=544, line=127 (Compiled frame; information may be imprecise)
 - sun.reflect.NativeMethodAccessorImpl.invoke0(java.lang.reflect.Method, java.lang.Object, java.lang.Object[]) @bci=0 (Interpreted frame)
 - sun.reflect.NativeMethodAccessorImpl.invoke(java.lang.Object, java.lang.Object[]) @bci=87 (Interpreted frame)
 - sun.reflect.DelegatingMethodAccessorImpl.invoke(java.lang.Object, java.lang.Object[]) @bci=6 (Interpreted frame)
 - java.lang.reflect.Method.invoke(java.lang.Object, java.lang.Object[]) @bci=57 (Interpreted frame)
 - ca.dmoj.java.SubmissionThread.run() @bci=53, line=27 (Interpreted frame)


Thread 19: (state = BLOCKED)
 - java.lang.Thread.sleep(long) @bci=0 (Interpreted frame)
 - ca.dmoj.java.ShockerThread.run() @bci=4, line=22 (Interpreted frame)


Thread 14: (state = BLOCKED)


Thread 13: (state = BLOCKED)


Thread 12: (state = BLOCKED)
 - java.lang.Object.wait(long) @bci=0 (Interpreted frame)
 - java.lang.ref.ReferenceQueue.remove(long) @bci=44 (Interpreted frame)
 - java.lang.ref.ReferenceQueue.remove() @bci=2 (Interpreted frame)
 - java.lang.ref.Finalizer$FinalizerThread.run() @bci=36 (Interpreted frame)


Thread 11: (state = BLOCKED)
 - java.lang.Object.wait(long) @bci=0 (Interpreted frame)
 - java.lang.Object.wait() @bci=2, line=503 (Interpreted frame)
 - java.lang.ref.Reference$ReferenceHandler.run() @bci=46 (Interpreted frame)


Thread 1: (state = BLOCKED)
 - java.lang.Object.wait(long) @bci=0 (Interpreted frame)
 - java.lang.Thread.join(long) @bci=38 (Interpreted frame)
 - java.lang.Thread.join() @bci=2 (Interpreted frame)
 - ca.dmoj.java.JavaSafeExecutor.main(java.lang.String[]) @bci=300, line=167 (Interpreted frame)
```

**Comment by Xyene** [comment]
This is still an issue on the latest judge; perhaps if the problematic frame were not compiled we would have better luck diagnosing the issue.

**Comment by Xyene** [comment]
Fixed along with #155: while the JVM does continue to hang, it is killed by the Python-side shocker.
